lib/rspec/support/differ.rb

```diff
@@ -116,6 +115,12 @@ def build_hunks(actual, expected)
         HunkGenerator.new(actual, expected).hunks
       end
 
+      def final_line(last_hunk)
+        output_message = last_hunk.diff(format_type).to_s.strip
+        # Checks last char of string for + or -, appends message on match
+        output_message.gsub(/(?<=[+-]$)\z/, "\\ No newline at end of input")
```

Comment:
I'd like to not have the trailing `+` unless it actually represents a proper `\n\n`, if you can't get quite that, feel free to push it up anyway I'll take another look :)